### ISSUE TITLE
How to transfer this method to 3D situation.

### ISSUE THREAD

**Issue by xubin04** [keyu-tian/SparK#74]
I want to know if I can effectively transfer this pre-trained approach to video prediction tasks. My data has dimensions of batch_size, time_stamp*channel, height, width. Can I mask different regions for different timestamps and can depth-wise processing handle this masking situation effectively.And thank you for your gerat work, this is a big step in CNN pretain method.

**Comment by keyu-tian** [comment]
Thanks @xubin04. Doing masked pretraining on video data is challenging but valuable. I think your depth-wise idea is worth trying. And you can also refer to [1] or [2] for their methods on ViT's video pretraining.

If you want to pretrain a 3D CNN-like network, I think our codebase can be used by you with only a few modifications (data processing, masking, etc.). Feel free to comment here again if you have questions.

---------------

[1] VideoMAE: Masked Autoencoders are Data-Efficient Learners for Self-Supervised Video Pre-Training
[2] Masked Autoencoders as spatiotemporal learners
